src/backend/distributed/commands/table.c

```diff
@@ -939,6 +960,34 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 				 */
 				constraint->skip_validation = true;
 			}
+			else if (constraint->contype == CONSTR_PRIMARY)
+			{
+				if (constraint->conname == NULL)
+				{
+					bool primary = true;
+					bool isconstraint = true;
+
+					Relation rel = RelationIdGetRelation(leftRelationId);
+
+					/*
+					 * Change the alterTableCommand directly so that the standard utility
+					 * hook runs it with the name we created.
+					 */
+					constraint->conname = ChooseIndexName(RelationGetRelationName(rel),
+														  RelationGetNamespace(rel),
+														  NULL, NULL, primary,
+														  isconstraint);
+					RelationClose(rel);
+
+					/*
+					 * Convert ALTER TABLE ... ADD PRIMARY ... command into
+					 * ALTER TABLE ... ADD CONSTRAINT <conname> PRIMARY KEY ...
+					 * and send it to the workers.
+					 */
+					return PreprocessAlterTableAddPrimaryKey(alterTableStatement,
```

Comment:
I like this idea of moving everything to a separate preprocess function like `PreprocessAlterTableAddPrimaryKey`. It is certainly cleaner, and we can also extend this Preprocess function in the future for other no-name constraint cases. ❤️ 

I think you can move everything to that function, i.e., you can generate the new name and update the alter table statement inside the preprocess function as well.

Reply:
Done.